onnx/defs/nn/defs.cc

```diff
@@ -2487,4 +2487,201 @@ ONNX_OPERATOR_SET_SCHEMA(
         )ONNX"
         ));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+  // we need at least three inputs to have a shape for this inference.
+  if (!hasNInputShapes(ctx, 3)) {
+    fail_shape_inference("Col2Im requires 3 inputs");
+  }
+
+  // validating 'input' input
+  auto input_shape = ctx.getInputType(0)->tensor_type().shape();
+  if ((input_shape.dim_size() < 2) || (input_shape.dim_size() > 3)) {
+    fail_shape_inference("input tensor must have 2 or 3 dimensions");
+  }
+  size_t n_input_dims = static_cast<size_t>(input_shape.dim_size());
+  if (input_shape.dim_size() > 2) {
+    // ignore first dim (batch axis) when there are three dims.
+    --n_input_dims;
+  }
+
+  // validating 'output_shape' input
+  std::vector<int64_t> output_shape;
+  const TensorProto* temp_output_shape = ctx.getInputData(1);
+  if (temp_output_shape) {
+    auto target_shape = ParseData<int64_t>(temp_output_shape);
+    for (auto val : target_shape) {
+      output_shape.push_back(val);
+    }
+    if (output_shape.size() < 2 || output_shape.size() > 3) {
+      fail_shape_inference("output_shape tensor must have 2 or 3 dimensions");
```

Comment:
the error message can be more clear. May be something like : "output_shape tensor contains partial shape for output. It must contain either 2 or 3 elements"